packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -0,0 +1,343 @@
+        this.#oncloseOriginal = callback;
+        this.#socket.onclose = this.#createWrapped(callback);
+      }
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onerror(): ((this: WebSocket, ev: Event) => any) | null {
+      return this.#onerrorOriginal;
+    }
+
+    set onerror(callback: ((this: WebSocket, ev: Event) => any) | null) {
+      this.#onerrorOriginal = callback;
+      this.#socket.onerror = this.#createWrapped(callback);
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onmessage(): ((this: WebSocket, ev: MessageEvent<any>) => any) | null {
+      return this.#onmessageOriginal;
+    }
+
+    set onmessage(
+      callback: ((this: WebSocket, ev: MessageEvent<any>) => any) | null,
+    ) {
+      this.#onmessageOriginal = callback;
+      this.#socket.onmessage = this.#createWrapped(callback);
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onopen(): ((this: WebSocket, ev: Event) => any) | null {
+      return this.#onopenOriginal;
+    }
+
+    set onopen(callback: ((this: WebSocket, ev: Event) => any) | null) {
+      this.#onopenOriginal = callback;
+      this.#socket.onopen = this.#createWrapped(callback);
+    }
+
+    close(code?: number, reason?: string): void {
+      this.#socket.close(code, reason);
+    }
+
+    send(data: string | Blob | ArrayBufferView | ArrayBufferLike): void {
+      this.#socket.send(data);
+    }
+
+    get CLOSED(): number {
+      return this.#socket.CLOSED;
+    }
+
+    get CLOSING(): number {
+      return this.#socket.CLOSING;
+    }
+
+    get CONNECTING(): number {
+      return this.#socket.CONNECTING;
+    }
+
+    get OPEN(): number {
+      return this.#socket.OPEN;
+    }
+
+    get binaryType(): BinaryType {
+      return this.#socket.binaryType;
+    }
+
+    set binaryType(value: BinaryType) {
+      this.#socket.binaryType = value;
+    }
+
+    get bufferedAmount(): number {
+      return this.#socket.bufferedAmount ?? 0;
+    }
+
+    get extensions(): string {
+      return this.#socket.extensions ?? '';
+    }
+
+    get protocol(): string {
+      return this.#socket.protocol;
+    }
+
+    get readyState(): number {
+      return this.#socket.readyState;
+    }
+
+    get url(): string {
+      return this.#socket.url;
+    }
+
+    addEventListener<K extends keyof WebSocketEventMap>(
+      type: K,
+      listener: (this: WebSocket, ev: WebSocketEventMap[K]) => any,
+      options?: boolean | AddEventListenerOptions,
+    ): void;
+
+    addEventListener(
+      type: string,
+      listener: EventListenerOrEventListenerObject,
+      options?: boolean | AddEventListenerOptions,
+    ): void;
+
+    addEventListener(type: any, listener: any, options?: any): void {
+      if (this.#isTornDown !== true) {
```

Comment:
1. Having only one return location is less error prone and easier to read compared to returning from multiple places.
2. Since JS is dynamically typed and many values can be truthy I prefer to be explicit that it should be a boolean.